### DISCUSSION TITLE
MultiMIDI_Interface

### DISCUSSION THREAD

**Discussion by santta** [tttapa/Control-Surface#447] [Q&A]
Does your library support MultiMIDI_Interface for simultaneous operation on USB and Serial ports? If it works, please explain in more detail how to use this function or give a link to the code with an example of use. Thanks.

I am building a midi footswitch to control my guitar processor via 5din (in rehearsals), I also want to use it at home to control AmpliTube on my PC via USB. Therefore, I need a code to simultaneously send midi program change commands over two streams (USB and Serial)

Arduino Leonardo using

**Comment by santta** [comment]
My test working sketch with simultaneous sending of midi via USB and DIN5

[sketchhhh.txt](https://github.com/tttapa/Control-Surface/files/6278656/sketchhhh.txt)

**Comment by tttapa** [comment]
For future reference, see https://tttapa.github.io/Control-Surface-doc/Doxygen/d5/da5/Dual-MIDI-Interface_8ino-example.html.

**Reply by santta** [reply]
thanks, I did it thanks to this example)))
